File: Programmers/Lv1/둘만의 암호.js
// 둘만의 암호, Lv1, JS 풀이

function solution(s, skip, index) {
    var answer = '';
    let arr = s.split('');

    arr.map((word) => {
        let tmp = word.charCodeAt();
        for(let i = 0; i < index; i++){
            tmp++;

            if(tmp > 122)
                tmp = 97;

            while(skip.includes(String.fromCodePoint(tmp))){
                tmp++;
                if(tmp > 122)
                    tmp = 97;

            }
        }

        answer = answer + String.fromCodePoint(tmp);

    });


    return answer;
}

File: Programmers/Lv1/둘만의 암호.py
# 둘만의 암호, Lv 1
def solution(s, skip, index):
    answer = ''
    # s의 각 문자를 index만큼 증가시키는데, skip의 값들은 넘어가기

    for w in s:

        tmp = ord(w)
        cnt = index
        while cnt > 0:
            tmp += 1
            if tmp > ord('z'):
                tmp = ord('a')
            if chr(tmp) in skip:
                cnt += 1  # 한번 더 가면 됨
            cnt -= 1

        answer += chr(tmp)
    return answer


